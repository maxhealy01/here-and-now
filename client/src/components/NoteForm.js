import React, { useState } from "react";
import { useMutation } from "@apollo/react-hooks";
import { ADD_NOTE } from "../utils/mutations";

const NoteForm = ({ lat, long }) => {
	const [noteText, setText] = useState("");
	const [addNote] = useMutation(ADD_NOTE);

	const handleChange = (event) => {
		setText(event.target.value);
	};

	const handleFormSubmit = async (event) => {
		event.preventDefault();

		try {
			await addNote({
				variables: { text: noteText, latitude: lat, longitude: long },
			});
			setText("");
			window.location.reload(false);
		} catch (e) {
			console.error(e);
		}
	};

	return (
		<div>
			<form
				className="flex-row justify-center justify-space-between-md align-stretch"
				onSubmit={handleFormSubmit}
			>
				<textarea
					placeholder="Write a note that will appear on the map!"
					value={noteText}
					className="form-input col-12 col-md-9"
					onChange={handleChange}
				/>
				<button className="btn col-12 col-md-3" type="submit">
					Submit
				</button>
			</form>
		</div>
	);
};

export default NoteForm;
