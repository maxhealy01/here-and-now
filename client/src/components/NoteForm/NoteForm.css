.formContainer {
	width: 75vw;
	margin: auto;
}

.button {
	display: block;
	margin: 0 auto;
}
.noteTextarea {
	margin-top: 6px;
	width: 75vw;
	/* height: 5rem; */
}
#rte {
	background-color: #cfb259;
	border: 1px var(--dark) solid;
}
.quill {
	background-color: var(--tertiary);
	border: 1px var(--dark) solid;
	border-top: none;
	color: var(--dark);
}

@media screen and (max-width: 768px) {
	.formContainer {
		width: 90vw;
		margin: auto;
	}
}
