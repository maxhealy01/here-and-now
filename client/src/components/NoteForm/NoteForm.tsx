import React, { useState } from "react";
import { FetchResult, useMutation } from "@apollo/client";
import { ADD_NOTE } from "../../utils/mutations";
import "./NoteForm.css";
import { RichTextEditor } from "@mantine/rte";
import { NoteType as Note } from "../../utils/typeDefs";

type NoteFormProps = {
	lat: number;
	long: number;
	setNotes: (notes: Note[]) => void;
	notes: Note[];
};
const NoteForm: React.FC<NoteFormProps> = ({ lat, long, setNotes, notes }) => {
	const [noteText, onChange] = useState("");
	const [addNote] = useMutation<Note>(ADD_NOTE);

	const handleFormSubmit: React.FormEventHandler<HTMLFormElement> = async (
		event
	) => {
		event.preventDefault();
		try {
			let note:
				| Note
				| FetchResult<
						Record<string, any>,
						Record<string, any>,
						Record<string, any>
				  > = await addNote({
				variables: {
					text: noteText,
					latitude: lat,
					longitude: long,
				},
			});

			setNotes([...notes, note?.data?.addNote]);
			onChange("");
		} catch (e) {
			console.error(e);
		}
	};
	return (
		<div className="formContainer">
			<form onSubmit={handleFormSubmit}>
				<div className="noteTextarea"></div>
				<RichTextEditor
					data-testid="text-editor"
					id="rte"
					controls={[
						["bold", "italic", "underline"],
						["unorderedList", "h1", "h2", "h3"],
						["sup", "sub"],
						["alignLeft", "alignCenter", "alignRight"],
					]}
					placeholder="Write a note that will appear on the map!"
					value={noteText}
					onChange={onChange}
				/>
				<div className="center">
					<button type="submit">Post</button>
				</div>
			</form>
		</div>
	);
};

export default NoteForm;
