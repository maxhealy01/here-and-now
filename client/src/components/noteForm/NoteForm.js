import React, { useState } from "react";
import { useMutation } from "@apollo/react-hooks";
import { ADD_NOTE } from "../../utils/mutations";
import "./NoteForm.css";

const NoteForm = ({ lat, long }) => {
	const [noteText, setText] = useState("");
	const [addNote] = useMutation(ADD_NOTE);

	const handleChange = (event) => {
		setText(event.target.value);
	};

	const handleFormSubmit = async (event) => {
		event.preventDefault();

		try {
			await addNote({
				variables: { text: noteText, latitude: lat, longitude: long },
			});
			setText("");
			window.location.reload(false);
		} catch (e) {
			console.error(e);
		}
	};

	return (
		<form onSubmit={handleFormSubmit}>
			<textarea
				className="noteTextarea"
				placeholder="Write a note that will appear on the map!"
				value={noteText}
				onChange={handleChange}
			/>
			<button type="submit">Submit</button>
		</form>
	);
};

export default NoteForm;
