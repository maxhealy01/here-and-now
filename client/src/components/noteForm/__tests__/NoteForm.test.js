import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import NoteForm from "../NoteForm";
import { ApolloProvider } from "@apollo/react-hooks";
import ApolloClient from "apollo-boost";

afterEach(cleanup);

const client = new ApolloClient({
	request: (operation) => {
		// we'll need the logic for setting id_token -- related to auth, both in client and server
		const token = localStorage.getItem("id_token");
		operation.setContext({
			headers: {
				authorization: token ? `Bearer ${token}` : "",
			},
		});
	},
	uri: "http://localhost:3001/graphql",
});

const lat = 30.267153;
const long = -97.743057;

describe("NoteForm component", () => {
	it("renders", () => {
		render(
			<ApolloProvider client={client}>
				<NoteForm lat={lat} long={long} />
			</ApolloProvider>
		);
	});

	it("matches snapshot DOM node structure", () => {
		const { asFragment } = render(
			<ApolloProvider client={client}>
				<NoteForm lat={lat} long={long} />
			</ApolloProvider>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
