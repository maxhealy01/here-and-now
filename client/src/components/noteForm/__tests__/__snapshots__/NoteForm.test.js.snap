// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteForm component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="formContainer"
  >
    <form>
      <textarea
        class="noteTextarea"
        placeholder="Write a note that will appear on the map!"
      />
      <button
        type="submit"
      >
        Post
      </button>
    </form>
  </div>
</DocumentFragment>
`;
